application/controllers/RegisterController.php

```diff
@@ -333,6 +346,12 @@ public function getTokenId($iSurveyId)
                 $this->aRegisterErrors[] = gT("This email address is already registered but email to that adress could not be delivered.");
             } else {
                 $this->sMailMessage = gT("The address you have entered is already registered. An email has been sent to this address with a link that gives you access to the survey.");
+                $now = dateShift(date("Y-m-d H:i:s"), "Y-m-d H:i", Yii::app()->getConfig('timeadjust'));
+                $delay = Yii::app()->getConfig('registrationEmailDelay');
+                if (!empty($oToken->sent) && $oToken->sent != "N" && dateShift($oToken->sent, "Y-m-d H:i", $delay) > $now) {
+                    /** @todo: Figure out a way to show a different message to the user */
+                    $this->sendRegistrationEmail = false;
```

Comment:
Why not update [https://github.com/LimeSurvey/LimeSurvey/blob/3c9909c1d2e983aa82fbdfd7f25753e783376966/application/controllers/RegisterController.php#L42C14-L42C22 ?](https://github.com/LimeSurvey/LimeSurvey/blob/3c9909c1d2e983aa82fbdfd7f25753e783376966/application/controllers/RegisterController.php#L46) or https://github.com/LimeSurvey/LimeSurvey/blob/3c9909c1d2e983aa82fbdfd7f25753e783376966/application/controllers/RegisterController.php#L42C14-L42C22

I think we can returen true but with with a different message https://github.com/LimeSurvey/LimeSurvey/blob/3c9909c1d2e983aa82fbdfd7f25753e783376966/application/controllers/RegisterController.php#L315

Reply:
It's done 3 lines before, just update it 
https://github.com/LimeSurvey/LimeSurvey/blob/3c9909c1d2e983aa82fbdfd7f25753e783376966/application/controllers/RegisterController.php#L348

Reply:
Yes, but from what we saw, that is never used later on the twigs

Reply:
Issue when moving from PHP to twig ?

sMessage was sent here : https://github.com/LimeSurvey/LimeSurvey/blob/9e43df218a8f5a37938fad50d95179eceafddbf1/application/controllers/RegisterController.php#L432

Maybe repoart as another issue , "The address you have entered is already registered. An email has been sent to this address with a link that gives you access to the survey." didn't appear too ?

Reply:
OK, sMessage was not used : https://github.com/LimeSurvey/LimeSurvey/blob/master/themes/survey/vanilla/views/subviews/registration/register_message.twig

Then user don't see it ....